File: src/JsonServices.Core.Tests/Messages/EventBroadcaster.cs
namespace JsonServices.Tests.Messages
{
	public class EventBroadcaster : IReturnVoid
	{
		public const string AfterStartupEventName = "IFoo.AfterStartup";

		public const string BeforeShutdownEventName = "IBar.BeforeShutdown";

		public const string FilteredEventName = "FilteredEvent";

		public string EventName { get; set; }

		public string StringArgument { get; set; }

		public bool BoolArgument { get; set; }

		public int IntArgument { get; set; }
	}
}

File: sample/sample-client/src/Messages/EventBroadcaster.ts
import { IReturnVoid } from 'json-services-client'

export class EventBroadcaster implements IReturnVoid {
    public EventName!: string
    public StringArgument!: string // for eventName === "FilteredEvent"
    public getTypeName() {
        return "JsonServices.Tests.Messages.EventBroadcaster"
    }
}